tests/test_signer.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+
+"""Test cases for 'signer.py'. """
+
+import sys
+import unittest
+
+import unittest
+import securesystemslib.formats
+import securesystemslib.keys as KEYS
+from securesystemslib.exceptions import FormatError, UnsupportedAlgorithmError
+
+# TODO: Remove case handling when fully dropping support for versions >= 3.6
+IS_PY_VERSION_SUPPORTED = sys.version_info >= (3, 6)
+
+# Use setUpModule to tell unittest runner to skip this test module gracefully.
+def setUpModule():
+    if not IS_PY_VERSION_SUPPORTED:
+        raise unittest.SkipTest('requires Python 3.6 or higher')
+
+# Since setUpModule is called after imports we need to import conditionally.
+if IS_PY_VERSION_SUPPORTED:
+    from securesystemslib.signer import Signature, SSlibSigner
+
+
+class TestSSlibSigner(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.rsakey_dict = KEYS.generate_rsa_key()
+        cls.ed25519key_dict = KEYS.generate_ed25519_key()
+        cls.ecdsakey_dict = KEYS.generate_ecdsa_key()
+        cls.DATA_STR = 'SOME DATA REQUIRING AUTHENTICITY.'
+        cls.DATA = securesystemslib.formats.encode_canonical(
+                cls.DATA_STR).encode('utf-8')
+
+
+    def test_sslib_sign(self):
+        dicts = [self.rsakey_dict, self.ecdsakey_dict, self.ed25519key_dict]
+        for scheme_dict in dicts:
+            # Test generation of signatures.
+            sslib_signer = SSlibSigner(scheme_dict)
+            sig_obj = sslib_signer.sign(self.DATA)
+
+            # Verify signature
+            verified = KEYS.verify_signature(scheme_dict, sig_obj.to_dict(),
+                    self.DATA)
+            self.assertTrue(verified, "Incorrect signature.")
+
+            # Removing private key from 'scheme_dict' - should raise a ValueError.
+            private = scheme_dict['keyval']['private']
+            scheme_dict['keyval']['private'] = ''
+            sslib_signer.key_dict = scheme_dict
+
+            with self.assertRaises((ValueError, FormatError)):
```

Comment:
Above you say:
> Removing private key from 'scheme_dict' - should raise a ValueError

Why do you assert for both `(ValueError, FormatError)`? Is this different for different keys? 🤦

Reply:
Yes, sadly.
For `rsa` keys   if there is no private key an `ValueError` is raised, 
for `ecdsa` and `ed25519` is a `FormatError`.

I will fix the comment above which says there should be a `ValueError`.

Reply:
> I was thinking should I create an issue or pr for this, but given that I am researching how to replaces the securesystemslib schemas probably it doesn't make sense.

I agree. I'll still make a note on #191.